src/FsAutoComplete.Core/FakeSupport.fs

```diff
@@ -44,4 +132,130 @@ module FakeSupport =
+        |> Seq.toArray
+      return targets
+  }
+
+  let private getTargetsJson (file:string) (ctx:FakeContext) : Async<Target []> = async {
+    // with --write-info support
+    let! rt = getFakeRuntime ()
+    let lines = ResizeArray<_>()
+    let fileName = Path.GetFileName file
+    let resultsFile = Path.GetTempFileName()
+    try
+      let workingDir = Path.GetDirectoryName file
+      let fakeArgs = sprintf "-s run --fsiargs \"--debug:portable --optimize-\" \"%s\" -- --write-info \"%s\"" fileName resultsFile
+      let args = sprintf "\"%s\" %s" rt fakeArgs
+      let exitCode, _ = Utils.runProcess (lines.Add) workingDir ctx.DotNetRuntime args
+      if exitCode <> 0 then
+        return [| errorTarget file (sprintf "Running Script 'fake %s' failed (%d)" fakeArgs exitCode) "We tried to retrieve the targets but your script failed" |]
+      else
+        let jsonStr = File.ReadAllText resultsFile
+        let jobj = JObject.Parse jsonStr
+
+        let parseDecl (t:JToken) =
+            { File = string t.["file"]; Line = int t.["line"]; Column = int t.["column"] }
+        let parseDep (t:JToken) =
+            { Name = string t.["name"]; Declaration = parseDecl t.["declaration"] }
+        let parseArray parseItem (a:JToken) =
+            (a :?> JArray)
+            |> Seq.map parseItem
+            |> Seq.toArray
+        let parseTarget (t:JToken) =
+            { Name = string t.["name"]
+              Declaration = parseDecl t.["declaration"]
+              HardDependencies = parseArray parseDep t.["hardDependencies"]
+              SoftDependencies = parseArray parseDep t.["softDependencies"]
+              Description = string t.["description"] }
+        let jTargets = jobj.["targets"] :?> JArray
+        let targets =
+          jTargets
+          |> Seq.map parseTarget
+          |> Seq.toArray
+        return targets
+    finally
+      try File.Delete resultsFile with e -> ()     
+  }      
+
+  let private getTargetsVersion (context:Runners.FakeContext) : Version option =
+
+    // soon there will be public apis and getProp can be removed...
+    let getProp (n:string) (t:obj) : 't =
+      let bf = System.Reflection.BindingFlags.Instance ||| System.Reflection.BindingFlags.NonPublic
+      t.GetType().GetProperty(n, bf).GetMethod.Invoke(t, [||]) :?> 't
+    let runtimeDeps : Runners.AssemblyInfo list = getProp "RuntimeDependencies" context.Config.RuntimeOptions
+    let targetVersion =
+      runtimeDeps
+      |> Seq.map (fun r -> System.Reflection.AssemblyName(r.FullName))
+      |> Seq.tryFind (fun a -> a.Name = "Fake.Core.Target")
+      |> Option.map (fun a -> a.Version)
+    targetVersion
+
+  let getTargets (file:string) (ctx:FakeContext) : Async<Target []> = async {
+    setupLogging()
+    match detectFakeScript file with
+    | None ->
+      return [| errorTarget file "not a FAKE 5 script" "This file is not a valid FAKE 5 script" |]
```

Comment:
maybe not the nicest way to report errors, given that you can probably "debug" this entry right now ...

Reply:
ideas?

Reply:
Use Result type to return from this function and handle error messages in the communication layer?

Reply:
Especially you can transform Error case into `CoreResponse.InfoRes` or `CoreResponse.ErrorRes`  in Commands.fs

Reply:
Ok I'm in the process of looking into this. Problem is: Does this allow me to add the error to the outline view as well? I guess on the client side I'd check for ErrorRes as well?

Is it OK to just extend the data-type and handle it myself? Is there any advantage of using `ErrorRes`?

Reply:
> Is it OK to just extend the data-type and handle it myself?

Yes, go for it if you need custom handling in UI

Reply:
Ok Done.